data-prepper-plugins/aws-plugin/src/main/java/org/opensearch/dataprepper/plugins/aws/AwsSecretManagerConfiguration.java

```diff
@@ -0,0 +1,104 @@
+package org.opensearch.dataprepper.plugins.aws;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import software.amazon.awssdk.arns.Arn;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest;
+import software.amazon.awssdk.services.sts.StsClient;
+import software.amazon.awssdk.services.sts.auth.StsAssumeRoleCredentialsProvider;
+import software.amazon.awssdk.services.sts.model.AssumeRoleRequest;
+
+import java.util.Optional;
+import java.util.UUID;
+
+public class AwsSecretManagerConfiguration {
+    private static final String DEFAULT_AWS_REGION = "us-east-1";
+    private static final String AWS_IAM_ROLE = "role";
+    private static final String AWS_IAM = "iam";
+
+    @JsonProperty("name")
+    @NotNull
+    @Size(min = 1, max = 512, message = "awsSecretName length should be between 1 and 512 characters")
+    private String awsSecretName;
+
+    @JsonProperty("region")
+    @Size(min = 1, message = "Region cannot be empty string")
+    private String awsRegion = DEFAULT_AWS_REGION;
+
+    @JsonProperty("sts_role_arn")
+    @Size(min = 20, max = 2048, message = "awsStsRoleArn length should be between 1 and 2048 characters")
+    private String awsStsRoleArn;
+
+    public String getAwsSecretName() {
+        return awsSecretName;
+    }
+
+    public Region getAwsRegion() {
+        return Region.of(awsRegion);
+    }
+
+    public SecretsManagerClient createSecretManagerClient() {
+        return SecretsManagerClient.builder()
+                .credentialsProvider(authenticateAwsConfiguration())
+                .region(getAwsRegion())
+                .build();
+    }
+
+    public GetSecretValueRequest createGetSecretValueRequest() {
+        return GetSecretValueRequest.builder()
+                .secretId(awsSecretName)
+                .build();
+    }
+
+    private AwsCredentialsProvider authenticateAwsConfiguration() {
```

Comment:
Let's use the existing AWSCredentialsProvider as provided by the AWS Plugin.

Reply:
https://github.com/opensearch-project/data-prepper/issues/3350